Wdb 60/setup video cards

kibartas opened the pull request:
![image](https://user-images.githubusercontent.com/37108355/117582784-3785f700-b0f3-11eb-9df0-498f5cd260a6.png)
![image](https://user-images.githubusercontent.com/37108355/117583035-7b2d3080-b0f4-11eb-9c09-112e7efd3360.png)
![image](https://user-images.githubusercontent.com/37108355/117585367-9fdbd500-b101-11eb-8c0d-82492325346d.png)

Commit:
[WDB-60]
LibraryPage now holds cards, if there are any. Make a mockedCards array in LibraryPage. Create onCardSelect handlers

Commit:
[WDB-60]
create VideoCard and VideoCardsByDate containers

Commit:
[WDB-60]
create VideoCard component

Commit:
[WDB-60]
merge master

Commit:
[WDB-60]
remove unused imports and highlight card on select

AArlauskas commented on backend/backend/appsettings.json:
```diff
@@ -1,6 +1,6 @@
 {
   "ConnectionStrings": {
-    "BackendContext": "Server=.\\SQLEXPRESS;Database=WeDontByte;Trusted_Connection=True;"
```
I don't think you are suppose to push these for non-docker environment

AArlauskas commented on backend/backend/appsettings.json:
```diff
@@ -12,7 +12,7 @@
   "EmailConfiguration": {
     "EmailAddress": "wedontbyte.team@gmail.com",
-    "EmailPassword": ""
+    "EmailPassword": "scynuekimkwfonzc"
```
This should not be pushed

AArlauskas commented on frontend/src/components/VideoCard/VideoCard.js:
```diff
@@ -0,0 +1,59 @@
+      <CardMedia
+        component="img"
+        style={{
+          backgroundColor: 'rgba(0,0,0,0.12)',
```
If you already have a styles.css file maybe you should consider moving these ones there since they are static

AArlauskas commented on frontend/src/containers/LibraryPage/LibraryPage.js:
```diff
@@ -1,61 +1,169 @@
 import './styles.css';
+import VideoCardsByDate from '../../components/VideoCardsByDate/VideoCardsByDate';
+
+const mockedCards = [
```
Not sure if this is where you want to keep the mocked data, maybe move it to the constants. Either way, it's temporary, so not that important

kibartas replied:
Yeah, I think it doesn't matter

AArlauskas commented on frontend/src/containers/LibraryPage/LibraryPage.js:
```diff
@@ -1,61 +1,169 @@
           onClose={this.handleUploadModalClose}
         />
-        <Grid item>
+        <Grid item style={{ flexGrow: 1 }}>
```
same comment for the styles

Commit:
[WDB-60]
revert changes to appsettings.json

Commit:
[WDB-60]
move inline styling to styles.css

Commit:
[WDB-60]
improve select logic, TopBar now shows how many videos are selected and lets you clear the selections

Commit:
[WDB-60]
change 'card' to 'video'

Commit:
[WDB-108]
implement card sorting

tmaciulis22 commented on frontend/src/components/VideoCard/VideoCard.js:
```diff
@@ -0,0 +1,48 @@
+        </Grid>
+      }
+      action={
+        <Grid item>
```
is Grid item necessary here if you don't have Grid container?

tmaciulis22 commented on frontend/src/components/VideoCard/VideoCard.js:
```diff
@@ -0,0 +1,48 @@
+      }}
+      disableTypography
+      title={
+        <Grid item>
```
is Grid item necessary here if you don't have Grid container?

tmaciulis22 commented on frontend/src/components/VideoCardsByDate/VideoCardsByDate.js:
```diff
@@ -0,0 +1,31 @@
+const VideoCardsByDate = ({ videoCards, onSelect, selectedCards }) => (
+  <Grid container item direction="column">
+    <Grid item>
+      <Typography variant="h6">{videoCards[0].date}</Typography>
```
```suggestion
      <Typography variant="h6">{videoCards[0].uploadDate}</Typography>
```

tmaciulis22 commented on frontend/src/components/VideoCard/styles.css:
```diff
@@ -0,0 +1,15 @@
+
+.card {
+    height: 300px;
+    width: 300px;
```
Maybe it would be reasonable to try `vh` values here? Since those values are adaptable to visible screen size

tmaciulis22 commented on frontend/src/containers/LibraryPage/LibraryPage.js:
```diff
@@ -1,61 +1,180 @@
+  {
+    id: '12',
+    title: 'Screaming guy',
+    date: '2020-11-23',
```
```suggestion
    uploadDate: '2020-11-23',
```

tmaciulis22 replied:
Below as well

tmaciulis22 reviewed (changes requested):
Minor stuff 🔍 Also, now maybe it's too late, but maybe Material UI's `GridList` or `ImageList`  components might have worked?
Overall good job, mate 👍🏻

tmaciulis22 commented on frontend/src/components/VideoCard/styles.css:
```diff
@@ -0,0 +1,15 @@
+.content {
```
You can ignore this suggestion, but here you can use BEM naming convention (block__elem--modifier) for css styles.  
Resource: BEM naming](http://getbem.com/naming/)  
So, in this case the names might be: `card`, `card__content`, `card__image`

AArlauskas commented on frontend/src/containers/LibraryPage/LibraryPage.js:
```diff
@@ -1,61 +1,180 @@
+          <Grid
+            className="cardContainer"
+            container
+            item
```
I believe this was already asked, but are you supposed to add both item and a container to a grid?

kibartas replied:
You can

Commit:
[WDB-60]
fix styling and naming and stuff

AArlauskas reviewed (approved):
LGTM ❤️

Commit:
Merge branch 'WDB-60/Setup-video-cards' into WDB-108/Sort-video-cards-by-upload-date

Commit:
[WDB-108]
bugfix: date -> uploadDate

Commit:
[WDB-108]
fix bug with sorting when firefox sorted and chrome didn't

kibartas committed:
Merge pull request #25 from kibartas/WDB-108/Sort-video-cards-by-upload-date

[WDB-108] Sort video cards by upload date
